pydm/widgets/shell_command.py

```diff
@@ -326,27 +342,69 @@ def redirectCommandOutput(self) -> bool:
         """
         Whether or not we should redirect the output of command to the shell.
 
-        Returns
-        -------
-        bool
+        This is deprecated in favor of the `stdout` property.
         """
-        return self._redirect_output
+        return self._stdout == TermOutputMode.SHOW
 
     @redirectCommandOutput.setter
     def redirectCommandOutput(self, value: bool) -> None:
+        self._uses_redirect_intf = True
+        if self._uses_stdout_intf:
+            logger.warning(
+                f"In PydmShellCommand widget with commands {self.commands}, "
```

Comment:
For the warning message in both setters, if the display was created with Designer self.commands might not yet be set and just print an empty list if it appears below `redirectCommandOutput` or `stdout/stderr` in the .ui file. Also possible with python code if setting commands late.

Reply:
I'll switch the warning to use the qt object name instead, which should always be set and should map to something the user can find in the designer going.